tempodb/encoding/vparquet4/rebatch_trace.go

```diff
@@ -0,0 +1,205 @@
+package vparquet4
+
+import (
+	"math"
+	"slices"
+	"strings"
+
+	"github.com/segmentio/fasthash/fnv1a"
+)
+
+// rebatchTrace removes redundant ResourceSpans and ScopeSpans from the trace through rebatching.
+func rebatchTrace(trace *Trace) {
+	if len(trace.ResourceSpans) == 0 {
+		return
+	}
+
+	// preallocate a map and a slice to collect the indices of unique ResourceSpans and ScopeSpans
+	uniqueIndexes := make([]int, 0, max(len(trace.ResourceSpans), len(trace.ResourceSpans[0].ScopeSpans)))
```

Comment:
took me a bit to figure out what you were using this. appreciate the effort on the in place alg